source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -355,7 +380,9 @@ absl::Status ProcessorState::handleBodyResponse(const BodyResponse& response) {
 
     // Send trailers if they are available and no data pending for processing.
     if (send_trailers_ && trailers_available_ && chunk_queue_.empty()) {
-      filter_.sendTrailers(*this, *trailers_);
+      if (body_mode_ != ProcessingMode::FULL_DUPLEX_STREAMED) {
```

Comment:
Same as comment above. Could please add some comments?

Or do we still keep chunk_queue_ there.  if not, `chunk_queue_.empty()` above doesn't apply.

Reply:
Done adding the comments.

Basically, for FULL_DUPLEX_STREAMED mode, if the trailers are available while waiting for body response, they were already sent here. So, no need to call this sending function again.

Reply:
Comment is not added?  

if trailer is already sent, should we update the status (like `trailers_available_` to false or clear the trailers_ ) when it is sent  to avoid sending it again, rather than checking the mode here

Reply:
Comments is added above:
// Skip doing this for FULL_DUPLEX_STREAMED mode as the trailers are already sent in this case.

trailers_available_ identify whether the trailers is seen by Envoy. Overload it will cause other issues.
trailers_ is the original trailers, which needs to be kept, as the server response is basically add, remove ,... against it.

Reply:
I still don't think checking the mode here makes sense. 

Basically we should have a state to indicate that trailer has been sent so that we don't need to send it again. This is a clearer and less error-prone approach. Whether you want to reuse existing fields or adding a new field is low-level details that I don't have strong preference.

Reply:
SG. Added a new state flag : trailers_sent_to_server_  to identify this.